pkg/storage/tsdb/caching_bucket.go

```diff
@@ -36,47 +62,133 @@ func (cfg *ChunksCacheConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix st
 	f.DurationVar(&cfg.SubrangeTTL, prefix+"subrange-ttl", 24*time.Hour, "TTL for caching individual chunks subranges.")
 }
 
-// Validate the config.
 func (cfg *ChunksCacheConfig) Validate() error {
-	if cfg.Backend != "" && cfg.Backend != string(storecache.MemcachedBucketCacheProvider) {
-		return fmt.Errorf("unsupported cache backend: %s", cfg.Backend)
+	return cfg.CacheBackend.Validate()
+}
+
+type MetadataCacheConfig struct {
+	CacheBackend
+
+	TenantsListTTL         time.Duration `json:"tenants_list_ttl"`
+	TenantBlocksListTTL    time.Duration `json:"tenant_blocks_list_ttl"`
+	ChunksListTTL          time.Duration `json:"chunks_list_ttl"`
+	MetafileExistsTTL      time.Duration `json:"metafile_exists_ttl"`
+	MetafileDoesntExistTTL time.Duration `json:"metafile_doesnt_exist_ttl"`
+	MetafileContentTTL     time.Duration `json:"metafile_content_ttl"`
+	MetafileMaxSize        int           `json:"metafile_max_size"`
```

Comment:
yaml, not json 😉

Reply:
Lol, I need to setup my IDE for adding `yaml`, instead of `json` :) Thanks for catching this.